app/controllers/generic_object_definition_controller.rb

```diff
@@ -286,4 +288,24 @@ def textual_group_list
   end
 
   helper_method :textual_group_list
+
+  def breadcrumbs_options
+    {
+      :breadcrumbs => [
+        {:title => _("Automation")},
+        {:title => _("Automate")},
+        {:title => _("Generic Objects")},
+      ],
+      :record_info => @generic_object_definition,
+    }
+  end
+
+  def build_breadcrumbs_from_tree
+    breadcrumbs = []
+    tree = build_tree if @tree.nil?
+    unless x_tree["active_node"].nil? || tree.nil?
+      breadcrumbs = current_tree_path(JSON.parse(tree.instance_variable_get("@bs_tree"))[0], x_tree["active_node"])
```

Comment:
You should not rely on `@bs_tree`, all the information is also in `nodes`, and `@bs_tree` is "temporary".

Reply:
(in general, you should also never use `instance_variable_get` in real code, just add a method to TreeBuilder)